Scripts/Engine/CustomBar.lua

```diff
@@ -0,0 +1,848 @@
+		local maxValue = LevelVars.Engine.CustomBars.bars[self.name].maxValue
+		local currentTarget = LevelVars.Engine.CustomBars.bars[self.name].targetValue or currentValue
+
+		-- Calculate new target value by adding the relative 'value' and clamp between 0 and 1000
+		local newTargetValue = math.max(0, math.min(maxValue, currentTarget + value))
+
+		-- Set the new target value
+		LevelVars.Engine.CustomBars.bars[self.name].targetValue = newTargetValue
+
+		-- Calculate total frames based on time and FPS (30 FPS)
+		local totalFrames = time * 30
+
+		-- Calculate the fixed interval for the entire transition
+		LevelVars.Engine.CustomBars.bars[self.name].fixedInterval = (newTargetValue - currentValue) / totalFrames
+	end
+end
+
+-- The function controls the visibility of a custom bar.
+-- 	@bool visible: true: Makes the bar visible.; false: Hides the bar.
+function CustomBar:SetVisibility(visible)
+    --the visible variable is a boolean
+	if LevelVars.Engine.CustomBars.bars[self.name] then
+		if visible and type(visible) == "boolean" then
+			LevelVars.Engine.CustomBars.bars[self.name].targetAlpha = 255
+			LevelVars.Engine.CustomBars.bars[self.name].visible = true
+		else
+			LevelVars.Engine.CustomBars.bars[self.name].targetAlpha = 0
+		end
+	end
+end
+
+--- The function checks whether a custom bar is currently visible.
+-- @treturn bool true if the bar is visible and false if it is not.
+function CustomBar:IsVisible()
+    
+	if LevelVars.Engine.CustomBars.bars[self.name] then
+		if LevelVars.Engine.CustomBars.bars[self.name].visible then
+			return true
+		else
+			return false
+		end
+	end
+end
+
+-- The function retrieves the current value of a custom bar.
+-- @treturn float returns the current value of a custom bar.
+function CustomBar:GetValue()
+	
+    if LevelVars.Engine.CustomBars.bars[self.name] then
+        return LevelVars.Engine.CustomBars.bars[self.name].oldValue
+	end
+end
+
+-- The function deletes all custom bars.
+CustomBar.DeleteAllBars = function ()
+	for _, customBar in pairs (LevelVars.Engine.CustomBars.bars) do
+		LevelVars.Engine.CustomBars.bars[customBar.name] = nil
+	end
+end
+
+-- This function prevents the creation of new health bars for enemies when set to false. However, it does not affect the health bars that have already been created.
+-- @bool value Specifies whether new health bars for enemies should be created. 
+CustomBar.ShowEnemiesHpGenericBar = function(value)
+	if type(value) == "boolean" then
+		LevelVars.Engine.CustomBars.EnemiesHpBar.status = value
+	end
+end
+
+-- The function deletes all the enemy health bars excluding those created by CustomBar.CreateEnemyHpBar.
+CustomBar.DeleteExistingHpGenericBars = function ()
+	for _, customBar in pairs (LevelVars.Engine.CustomBars.bars) do
+		if customBar.getActionType == 4 then
+			LevelVars.Engine.CustomBars.bars[customBar.name] = nil
+		end
+	end
+end
+
+---Set background properties

```

Comment:
This formatting is incorrect and produces incorrect documentation generation for ` SetBackgroundPosition` function.

Reply:
Done